internal/provider/data_source_pullzone.go

```diff
@@ -0,0 +1,86 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourcePullZone() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourcePullZoneRead,
+
+		Schema: map[string]*schema.Schema{
+			"pull_zone_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			keyAWSSigningKey: {
+				Type:        schema.TypeString,
+				Description: "AWS Signing Key",
+				Computed:    true,
+			},
+			keyAWSSigningRegionName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The AWS Signing region name.",
+			},
+			keyAWSSigningSecret: {
+				Type:        schema.TypeString,
+				Sensitive:   true,
+				Computed:    true,
+				Description: "The AWS Signing region secret.",
+			},
+			keyCnameDomain: {
+				Type:        schema.TypeString,
+				Description: "The CNAME domain of the Pull Zone for setting up custom hostnames.",
+				Computed:    true,
+			},
+			keyName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The name of the Pull Zone.",
+			},
+			keyStorageZoneID: {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The ID of the storage zone that the Pull Zone is linked to.",
+			},
+			keyZoneSecurityEnabled: {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "True if the URL secure token authentication security is enabled.",
+			},
+			keyZoneSecurityIncludeHashRemoteIP: {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "True if the zone security hash should include the remote IP.",
```

Comment:
Those are properties that, while exposed via the Video Library update endpoint, are not fetchable as part of the Video Library, as they reside exclusively on the Pull Zone. I put them in here as I think they'll be helpful in certain cases to determine the full state of a video library.

Reply:
Correct, those are pass throughs that bunny internally sets on the managed pull-zone. The unfortunate part is that they don't also return them in the video library API. I asked the Bunny team about this in a support request and they said you have to query the Pull Zone directly to get them. The other alternative to this is to do a special PullZone.Get as part of the VideoLibrary update process, which doesn't feel great to me. Exposing them in this way seemed fairly simple, although it has the downside of your `bunny_videolibrary` resource state potentially being incorrect if something weird happens.

I do like bunny a lot, but when they do stuff like this I get a bit frustrated with them.

Reply:
hum, I see, that makes it complicated.

Having to use fields in a pullzone datasource as workaround for the issue makes the Terraform provider difficult to use.
Users would need to know these API internals, that a videolib has an underlying pullzone, that some videolib attributes are stored in the pullzone and also the different attribute names.

A simple solution in other cases were a field was not retrievable via the Get endpoint was to mark the resource attribute as `ForceNew: true`.
I don't know if that is feasible for those videolib attributes.
Would be having to recreated the videolib when these attributes are changed an option?

Otherwise I think it is better to do the additional pullzone get call in the `ReadContext` function of the videolib.

Reply:
ForceNew isn't really an option. It would delete all of your video content. As of the current state of the videolibrary resource I did something similar to what we did on the custom_404 property for storage zones if you recall that. Effectively, if the API call succeeds we just update the state to whatever value was sent to bunny. Which seems finish... until I found a bug in the bunny API where they were silently failing to set those properties in certain circumstances (I filed a bug report with them and they're going to fix it).

But still, stuff like that can happen unless you do a hard read on them.

Reply:
Also, if we're going to read the pullzone in the bunny videolibrary resource, then it seems like we might as well return the entire data set that we care about from it instead of doing a datasource. The data source seemed like a cleaner way at first, but maybe that should be considered instead.

Reply:
> Also, if we're going to read the pullzone in the bunny videolibrary resource, then it seems like we might as well return the entire data set that we care about from it instead of doing a datasource. 

Let's go for that solution.
Hiding these ugly internals to make the provider easier to use will be worth the additional api requests and code.